### PR TITLE
Remove SyntaxKind.DataKeyword

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes https://github.com/dotnet/roslyn/issues/53589

### CODE REVIEW

**Comment by Youssef1313** [comment]
> .tools\msbuild\16.8.0-preview2.1\tools\Common7\IDE\CommonExtensions\Microsoft\NuGet\NuGet.RestoreEx.targets(10,5): error : (NETCORE_ENGINEERING_TELEMETRY=Restore) Failed to download package 'VsWebsite.Interop.8.0.50727' from 'https://pkgs.dev.azure.com/azure-public/3ccf6661-f8ce-4e8a-bb2e-eff943ddd3c7/_packaging/2a239fd0-3e21-40b0-b9d6-bc122fec7eb2/nuget/v3/flat2/vswebsite.interop/8.0.50727/vswebsite.interop.8.0.50727.nupkg'.
Response status code does not indicate success: 503 (Operations per second is over the account limit.).

File: src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs [hunk 1]
```diff
@@ -381,8 +381,9 @@ public enum SyntaxKind : ushort
         AndKeyword = 8439,
         /// <summary>Represents <see langword="not"/>.</summary>
         NotKeyword = 8440,
-        /// <summary>Represents <see langword="data"/>.</summary>
-        DataKeyword = 8441,
+
+        // open for future use.
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs:385]:
```suggestion
        // 8441 open for future use.
```

**Review by 333fred — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 5)

**Comment by jcouv** [comment]
Merged/squashed. Thanks @Youssef1313 !

File: src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs [hunk 2]
```diff
@@ -381,8 +381,9 @@ public enum SyntaxKind : ushort
         AndKeyword = 8439,
         /// <summary>Represents <see langword="not"/>.</summary>
         NotKeyword = 8440,
-        /// <summary>Represents <see langword="data"/>.</summary>
-        DataKeyword = 8441,
+
+        // 8441 open for future use.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxKind.cs:385]: >// 8441 open for future use. [](http://example.com/codeflow?start=8&length=28)

@333fred, @jcouv

I believe the decision of compat council was to reserve the number. I.e. prevent its reuse now or in the future. It doesn't look like this comment is in line with that.

**Pull request merged**